### DISCUSSION TITLE
Query type and strict-id-in-types

### DISCUSSION THREAD

**Discussion by ivandotv** [graphql-hive/graphql-eslint#871] [General]
When extending the recommended schema, and with this query type:

```graphql
type Query {
  me:User!
}
```
I get an `stric-id-in-types` error. Shouldn't this be disabled for `Query` and `Mutation` types?

**Comment by dotansimha** [comment]
@B2o5T I think we might need to ignore the root schemas on this rule?

**Reply by dimaMachina** [reply]
Exactly 🎉 !
I added a fix for ignoring root types, but now this rule will require schema for determining root type names (query, mutation, and subscription). The related tests were added also 🚀

**Comment by dimaMachina** [comment]
@ivandotv You are totally right, this was a mistake from the developer of this rule, this will be fixed on the next version of `graphql-eslint`.

**Comment by dotansimha** [comment]
Thanks @ivandotv ,
@B2o5T fixed it in https://github.com/dotansimha/graphql-eslint/pull/879 , I released it as `@graphql-eslint/eslint-plugin@3.6.0
`.
